lib/trento/infrastructure/software_updates/suma.ex

```diff
@@ -0,0 +1,188 @@
+        {:noreply, new_state}
+
+      {:error, reason} ->
+        {:stop, reason}
+    end
+  end
+
+  @impl true
+  def handle_info(
+        {{:get_system_id, fully_qualified_domain_name}, reply_to: from},
+        %State{
+          url: url,
+          auth: auth_cookie
+        } = state
+      ) do
+    GenServer.reply(from, get_system_id(url, auth_cookie, fully_qualified_domain_name))
+
+    {:noreply, state}
+  end
+
+  defp process_identifier(server_name), do: {:global, identificaton_tuple(server_name)}
+
+  defp identificaton_tuple(server_name), do: {__MODULE__, server_name}
+
+  defp setup_auth(%State{} = state) do
+    with {:ok, %{url: url, username: username, password: password, ca_cert: ca_cert}} <-
+           SoftwareUpdates.get_settings(),
+         credentials <- %{username: username, password: password},
+         api_url <- SumaApi.get_suma_api_url(url),
+         {:ok, auth_cookie} <- try_login(api_url, credentials, @login_retries) do
+      {:ok,
+       %State{
+         state
+         | url: api_url,
+           username: username,
+           password: password,
+           ca_cert: ca_cert,
+           auth: auth_cookie
+       }}
+    end
+  end
+
+  defp try_login(_, _, 0) do
+    Logger.error("Failed to Log into SUSE Manager. Max retries reached.")
+    {:error, :max_login_retries_reached}
+  end
+
+  defp try_login(url, %{username: username, password: password} = credentials, retry) do
+    case login(url, username, password) do
+      {:ok, _} = successful_login ->
+        successful_login
+
+      {:error, reason} ->
+        Logger.error("Failed to Log into SUSE Manager, retrying...", error: inspect(reason))
+        try_login(url, credentials, retry - 1)
+    end
+  end
+
+  defp login(url, username, password) do
+    login_response = client().login(url, username, password)
+
+    case login_response do
+      {:ok, %HTTPoison.Response{headers: headers, status_code: 200} = response} ->
+        Logger.debug("Successfully logged into suma #{inspect(response)}")
+        {:ok, SumaApi.get_session_cookies(headers)}
+
+      {:ok, %HTTPoison.Response{status_code: _} = response} ->
+        Logger.error(
+          "Failed to login to SUSE Manager due to unsuccessful response. Response: #{inspect(response)}"
+        )
+
+        {:error, :login_error}
+
+      {:error, reason} ->
+        Logger.error("Failed to login to SUSE Manager due to an error. Error: #{inspect(reason)}")
+        {:error, :login_error}
+    end
+  end
+
+  defp get_system_id(url, auth_cookie, fully_qualified_domain_name) do
+    with {:ok, %HTTPoison.Response{status_code: 200, body: body}} <-
+           client().get_system_id(url, auth_cookie, fully_qualified_domain_name),
+         {:ok, %{"success" => true, "result" => result}} <- Jason.decode(body),
+         {:ok, system_id} <- extract_system_id(result) do
+      {:ok, system_id}
+    else
+      _ = error ->
```

Comment:
You're actually using the error, no need for the black hole

```suggestion
      error ->
```